[TASK] Fixed patch and added it to the build process

dkdlniazman opened the pull request:
Fix for #12 

* Added 'compile-patch-nutch' as dependency to 'nutch-compile'
* Added missing imports to patch file

dkdlniazman committed:
[TASK] Restructure and update README.md

Restructures and updates the README.md, also adds bullet points
under "Development Environment Setup".

dkdlniazman committed:
[TASK] Fixed patch and added it to the build process

Fixes and adapts the patch, originating from nutch-585, for Apache
Nutch 1.19.

dkd-kaehm commented:
@dkdlniazman 
Please add github actions setup for build process.
The tests & co. we'll add at later point.

See:
* https://github.com/features/actions
* https://github.com/actions/setup-java

dkd-friedrich committed:
[BUGFIX] Remove existing Solr document id

Apache Solr for TYPO3 requires a special document id structure
that is build within the SiteHashIndexingFilter.

As in newer Nutch versions an id is already prepared and to
prevent that a multiple field is used for the id, an existing
id field is now removed before the custom id is added.

dkd-friedrich committed:
[BUGFIX] Remove invalid log4j option cmdstdout

The following error occured during crawling:
Could not find value for key log4j.appender.cmdstdout

This commit fixes this issue by removing the invalid option

dkd-friedrich committed:
[TASK] Update documentation for v2.2

Updates the documentation for Apache Nutch 1.19 and Apache Nutch
plugins for TYPO3 CMS version 2.2.

dkd-friedrich commented:
@dkdlniazman I looked at the patch and tested crawling with EXT:solr 11.2, 11.5 and main, a few adjustments were required

Please also look over the updates again

- Patch: 
  - Used current code from 1.19
  -  removed duplication of getNoFollow block
- fix for Document id in SiteHashIndexingFilter 
- Documentation updated

If everything works for you, I'll squash my patch customization and then merge the PR

dkdlniazman commented:
@dkd-friedrich Thanks for your improvements! I added my documentation updates from https://github.com/dkd/nutch-typo3-cms/pull/10

Everything else looks good to me, we can merge once the CI runs through
